### PR TITLE
Add Flag component 

### LINKED ISSUE #177: <Flag /> Component
http://semantic-ui.com/elements/flag.html

### PR DESCRIPTION
**Pull request by layershifter**
## API discussion

This component is pretty simple. There is only one changable property. I propose use `name` property as in `Icon` #279.

``` html
<i class="ae flag"></i>
<i class="france flag"></i>
<i class="myanmar flag"></i>
```

``` js
<Flag name='ae' />
<Flag name='france' />
<Flag name='myanmar' />
```

Feedback welcome :smile:

### CODE REVIEW

**Comment by layershifter** [comment]
Fixes #177

File: src/elements/Flag/Flag.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+function Flag(props) {
+  const { className, name, ...rest } = props
+  const classes = cx(
+    name,
+    className,
+    'flag',
+  )
+
+  return (<i className={classes} {...rest} />)
```
Comment (levithomason) [inline src/elements/Flag/Flag.js]: Parens are only required for multi line JSX.  They can be removed here.

File: src/elements/Flag/Flag.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  library: META.library.semanticUI,
+  name: 'Flag',
+  type: META.type.element,
+}
+
+Flag.propTypes = {
+  /** Classes to add to the Flag className. */
+  className: PropTypes.string,
+
+  /** Name for element, can use the two digit country code, the full name, or a common alias */
```
Comment (levithomason) [inline src/elements/Flag/Flag.js]: `Name of element` seems like it should be `Flag name`.

File: test/specs/elements/Flag/Flag-test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+describe('Flag', () => {
+  common.isConformant(Flag)
+
+  it('renders an <i /> element', () => {
+    shallow(<Flag />)
+      .should.have.tagName('i')
+  })
+
+  it('adds the "flag" class', () => {
```
Comment (levithomason) [inline test/specs/elements/Flag/Flag-test.js]: This test is covered by the `isConformant()` test.  The `isConformant()` test ensures the className includes has the component name.

**Comment by levithomason** [comment]
Looks good to me 👍 Let's add the flag names, ISO codes, and aliases in the prop validation for `name` to finish it off.

**Comment by layershifter** [comment]
>  Let's add the flag names, ISO codes, and aliases in the prop validation for name to finish it off.

Do you mean something like this?

``` js
const names = [
  'andorra',
  'ad',
  ...,
  'burma',
  'mm',
  'myanmar'
];

Flag._meta = {
  props: {
    name: names
  }
}

Flag.propTypes = {
 size: PropTypes.oneOf(Flag._meta.props.name)
}
```

**Comment by levithomason** [comment]
Yep, that will do it.  As far as syntax, you can list them out horizontally since there will be many.  We allow 120 columns per line.

**Comment by levithomason** [comment]
Great, lmk if there is anything else you'd like to do here.  Otherwise, I'll merge this once tests pass.

**Comment by levithomason** [comment]
Oh, almost forgot.  Go ahead and check it off in the README.md under support :)

**Comment by layershifter** [comment]
Updated branch and README :smile:

File: test/specs/elements/Flag/Flag-test.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+describe('Flag', () => {
+  common.isConformant(Flag)
+
+  it('renders an <i /> element', () => {
+    shallow(<Flag />)
+      .should.have.tagName('i')
+  })
+
+  it('passes property "name" to class', () => {
```
Comment (levithomason) [inline test/specs/elements/Flag/Flag-test.js]: Sorry I missed this earlier, this test should be replaced with the common tests for "value only" props:

``` js
common.propValueOnlyToClassName(Flag, 'name')
```

This test ensures the prop is defined in _meta, is added to className when it should be, and is not added when it shouldn't be.

**Comment by levithomason** [comment]
Thanks for the component!

**Pull request merged**